cmd/opentelemetry-collector/app/defaults/merge_test.go

```diff
@@ -0,0 +1,176 @@
+				},
+			},
+			"zipkin": &zipkinreceiver.Config{
+				ReceiverSettings: configmodels.ReceiverSettings{
+					Endpoint: "master_zipkin_url",
+				},
+			},
+		},
+		Processors: configmodels.Processors{
+			"attributes": &attributesprocessor.Config{
+				Actions: []attributesprocessor.ActionKeyValue{{Key: "foo"}},
+			},
+		},
+		Service: configmodels.Service{
+			Extensions: []string{"def", "master1", "master2"},
+			Pipelines: configmodels.Pipelines{
+				"traces": &configmodels.Pipeline{
+					Receivers:  []string{"jaeger", "zipkin"},
+					Processors: []string{"attributes"},
+				},
+				"traces/2": &configmodels.Pipeline{
+					Processors: []string{"example"},
+				},
+			},
+		},
+	}
+	expected := &configmodels.Config{
+		Receivers: configmodels.Receivers{
+			"jaeger": &jaegerreceiver.Config{
+				Protocols: map[string]*receiver.SecureReceiverSettings{
+					"grpc":           {ReceiverSettings: configmodels.ReceiverSettings{Endpoint: "master_jaeger_url", Disabled: true}},
+					"thrift_compact": {ReceiverSettings: configmodels.ReceiverSettings{Endpoint: "def"}},
+				},
+			},
+			"zipkin": &zipkinreceiver.Config{
+				ReceiverSettings: configmodels.ReceiverSettings{
+					Endpoint: "master_zipkin_url",
+				},
+			},
+		},
+		Processors: configmodels.Processors{
+			"batch": &batchprocessor.Config{
+				SendBatchSize: uint32(160),
+			},
+			"attributes": &attributesprocessor.Config{
+				Actions: []attributesprocessor.ActionKeyValue{{Key: "foo"}},
+			},
+		},
+		Service: configmodels.Service{
+			Extensions: []string{"def", "def2", "master1", "master2"},
+			Pipelines: configmodels.Pipelines{
+				"traces": &configmodels.Pipeline{
+					Receivers:  []string{"jaeger", "zipkin"},
+					Processors: []string{"batch", "attributes"},
+				},
+				"traces/2": &configmodels.Pipeline{
+					Processors: []string{"example"},
+				},
+			},
+		},
+	}
+	err := MergeConfigs(cfg, overrideCfg)
+	require.NoError(t, err)
+	assert.Equal(t, expected, cfg)
+}
+
+func TestMergeConfigFiles(t *testing.T) {
+	testFiles := []string{"emptyoverride", "addprocessor", "multiplecomponents"}
+	v, _ := config2.Viperize(elasticsearch.DefaultOptions().AddFlags)
+	cmpts := Components(v)
+	for _, f := range testFiles {
+		t.Run(f, func(t *testing.T) {
+			cfg, err := loadConfig(cmpts, fmt.Sprintf("testdata/%s.yaml", f))
+			require.NoError(t, err)
+			override, err := loadConfig(cmpts, fmt.Sprintf("testdata/%s-override.yaml", f))
+			require.NoError(t, err)
+			merged, err := loadConfig(cmpts, fmt.Sprintf("testdata/%s-merged.yaml", f))
+			require.NoError(t, err)
+			err = MergeConfigs(cfg, override)
+			require.NoError(t, err)
+			assert.Equal(t, merged, cfg)
+		})
+	}
+}
+
+func loadConfig(factories config.Factories, file string) (*configmodels.Config, error) {
+	// config.Load fails to load an empty config
+	if file == "testdata/emptyoverride-override.yaml" {
```

Comment:
If I'm reading this correctly the file is not part of this PR and this validation is just a way to prevent config.Load to fail and instead returns an empty config, to test empty configuration merges.